diag_manager/fms_diag_file_object.F90

```diff
@@ -751,9 +783,64 @@ subroutine open_diag_file(this, time_step)
     diag_file%next_open = diag_time_inc(diag_file%next_open, VERY_LARGE_FILE_FREQ, DIAG_DAYS)
   endif
 
-!TODO: closing the file here for now, just to see if it works
-  call close_file(diag_file%fileobj)
+  file_is_opened = .true.
+  domain => null()
+  diag_file => null()
 end subroutine open_diag_file
 
+!< @brief Writes the axis metadata for the file
+subroutine write_metadata(this, diag_axis)
+  class(fmsDiagFileContainer_type), intent(inout), target :: this            !< The file object
+  class(fmsDiagAxisContainer_type), intent(in)            :: diag_axis(:)    !< Diag_axis object
+
+  class(fmsDiagFile_type), pointer     :: diag_file      !< Diag_file object to open
+  class(FmsNetcdfFile_t),  pointer     :: fileobj        !< The fileobj to write to
+  integer                              :: i              !< For do loops
+  integer                              :: j              !< diag_file%axis_ids(i) (for less typing)
+  integer                              :: parent_axis_id !< Id of the parent_axis
+
+  diag_file => this%FMS_diag_file
+  fileobj => diag_file%fileobj
+
+  do i = 1, diag_file%number_of_axis
+    j = diag_file%axis_ids(i)
+    parent_axis_id = diag_axis(j)%axis%get_parent_axis_id()
+    if (parent_axis_id .eq. DIAG_NULL) then
+      call diag_axis(j)%axis%write_axis_metadata(fileobj)
+    else
+      call diag_axis(j)%axis%write_axis_metadata(fileobj, diag_axis(parent_axis_id)%axis)
+    endif
+  enddo
+
+end subroutine write_metadata
+
+!< @brief Writes the axis data for the file
+subroutine write_axis_data(this, diag_axis)
+  class(fmsDiagFileContainer_type), intent(inout), target :: this            !< The file object
+  class(fmsDiagAxisContainer_type), intent(in)            :: diag_axis(:)    !< Diag_axis object
+
+  class(fmsDiagFile_type), pointer     :: diag_file      !< Diag_file object to open
+  class(FmsNetcdfFile_t),  pointer     :: fileobj        !< The fileobj to write to
+  integer                              :: i              !< For do loops
+  integer                              :: j              !< diag_file%axis_ids(i) (for less typing)
+  integer                              :: parent_axis_id !< Id of the parent_axis
+
+  diag_file => this%FMS_diag_file
+  fileobj => diag_file%fileobj
+
+  do i = 1, diag_file%number_of_axis
+    j = diag_file%axis_ids(i)
+    parent_axis_id = diag_axis(j)%axis%get_parent_axis_id()
+    if (parent_axis_id .eq. DIAG_NULL) then
+      call diag_axis(j)%axis%write_axis_data(fileobj)
+    else
+      call diag_axis(j)%axis%write_axis_data(fileobj, diag_axis(parent_axis_id)%axis)
+    endif
+  enddo
+
+  !TODO: closing the file here for now, just to see if it works
+  call close_file(fileobj)
```

Comment:
oh boy...

Reply:
This gets cleaned up in the next PR, once all of the time stuff is set up "correctly"